syntax = "proto3";

package food;

service Supplier {
  rpc GetStores (ItemQuery) returns (StoreReply) {}
}

service Vendor {
  rpc GetItemInfo (ItemStoreQuery) returns (ItemInfoReply) {}
}

// Request containing queried item/ingredient.
message ItemQuery {
  string item = 1;
}

// Message containing stores with the requested item
message StoreReply {
  repeated string stores = 1;
}

// Request containing store and item
message ItemStoreQuery {
  string item = 1;
  string store = 2;
}


// Response containing price and inventory for given store and item
message ItemInfoReply {
  double price = 1;
  int32 inventory = 2;
}


// package .usr.local.google.home.janetvu.janet-starter-project.files.starter-proj;
